src/VS.Mefx/Commands/PartNode.cs

```diff
@@ -0,0 +1,145 @@
+        {
+            this.Part = definition;
+            this.IsWhiteListed = false;
+            this.VerboseMessages = new HashSet<string> { message };
+            this.ImportRejects = new HashSet<PartEdge>();
+            this.RejectsCaused = new HashSet<PartEdge>();
+            this.Level = currLevel;
+            this.IsWhiteListed = false;
+
+            this.ExportingContracts = new List<string>();
+            foreach (var export in this.Part.ExportDefinitions)
+            {
+                if (export.Key == null)
+                {
+                    continue;
+                }
+
+                this.ExportingContracts.Add(export.Value.ContractName);
+            }
+        }
+
+        /// <summary>
+        /// Gets the definition associated with the current part.
+        /// </summary>
+        public ComposablePartDefinition Part { get; private set; } // Represents the part associated with the current node
+
+        /// <summary>
+        /// Gets the verbose rejection messages associated with the current part.
+        /// </summary>
+        public HashSet<string> VerboseMessages { get; private set; } // Rejection Message(s) associated with the current part
+
+        /// <summary>
+        /// Gets the "children" of the current node, which represents parts that the current
+        /// node imports that have import issues themselves.
+        /// </summary>
+        public HashSet<PartEdge> ImportRejects { get; private set; }
+
+        /// <summary>
+        /// Gets the "parent" of the current node, which represents parts that the current
+        /// node caused import issues in due to its failure.
+        /// </summary>
+        public HashSet<PartEdge> RejectsCaused { get; private set; }
+
+        /// <summary>
+        /// Gets the level of the current node, which serves as an indicator
+        /// of its depth in the rejection stack.
+        /// </summary>
+        public int Level { get; private set; }
+
+        /// <summary>
+        /// Gets a value indicating whether the current node has been whitelisted by the user.
+        /// </summary>
+        public bool IsWhiteListed { get; private set; }
+
+        /// <summary>
+        /// Gets the name of contracts exported by the part other than itself.
+        /// </summary>
+        public List<string> ExportingContracts { get; private set; }
+
+        /// <summary>
+        /// Method to get the label for the node.
+        /// </summary>
+        /// <returns>The name of the associated part.</returns>
+        public string GetName()
+        {
+            return this.Part.Type.FullName!;
+        }
+
+        /// <summary>
+        /// Method to check if the given node imports any parts with import issues.
+        /// </summary>
+        /// <returns> A boolean indicating it the given node is a leaf node.</returns>
+        public bool IsLeafNode()
+        {
+            return this.ImportRejects.Count == 0;
```

Comment:
For these short one-liner methods, consider using the expression body syntax:

```cs
public bool IsLeafNode() => this.ImportsRejects.Count == 0;
```

And in fact this and some others may be better expressed as a property:

```cs
public bool IsLeafNode => this.ImportsRejects.Count == 0;
```